Create lesson: Inscribed Rectangle Problem (#68)
3b1b/3Blue1Brown.com: content/lessons/2016/inscribed-rectangle-problem/index.md

{{< section >}}

# The Inscribed Square Problem
No one knows the answer to the following question:
> If you have a closed curve in 2D space, will you always be able to find four points on this curve that make up a square?

This is known as the Toeplitz' conjecture. By "closed curve", we mean you squiggle some line through space in a potentially crazy way and end at the point where you started.

{{< figure image="inscribed-square.png" >}}
We can find four points on *this* closed loop that form a square. But can we do the same for *any* loop?
{{< /figure >}}

If your closed loop is a circle, for example, it’s quite easy to find an inscribed square. Infinitely many squares, in fact. If your loop was an ellipse, it’s still pretty easy to find an inscribed square.

{{< figure image="circle-and-ellipse.png" >}}
Finding inscribed squares in circles and ellipses is easy. But what about other, wilder loops?
{{< /figure >}}

This tweet shows a bunch of crazy loops, and at every frame there is indeed a square inscribed inside:

> {{< tweet 1390169453346787329 >}}

Author: The tweet component just comes built-in to Hugo, so I say we don't worry about it for now.

This mapping allows us to turn any pair of two points on the loop into a pair of two numbers between $0$ and $1$, which is much easier to reason about.

{{< figure image="point-pair-to-numbers.png" >}}
Every point on the loop corresponds to a number between $0$ and $1$. So a pair of points on the loop corresponds to a pair of numbers between $0$ and $1$.
{{< /figure >}}

We can place one of the two values on a new y-axis, using a second interval, so that each pair of values between $0$ and $1$ (and thus each pair of points on the loop) is associated with a single point in a $1 \times 1$ square.

{{< figure image="pair-to-square.png" >}}
By giving each of the two numbers (corresponding to our original two points on the loop) a unique axis to live on, we can represent any pair of points on the loop with a single point in a $1 \times 1$ square of 2D space.
{{< /figure >}}

Remember, we’re trying to find a surface that naturally represents the set of all pairs of points on the loop. This square is the first step to doing that.

**Stop to think:** If you have a pair of points on the loop, how do you find the corresponding point on the square?

> **Answer:** {{< spoiler >}}Convert each loop point to a number between 0 and 1, and then use those two numbers as the x and y coordinates in the square.{{< /spoiler >}}

Reviewer: ```suggestion
**Answer:** {{< spoiler >}} This assumes you've already chosen a _parameterization_ of the loop, meaning a continuous way to associate individual points with numbers between 0 and 1. From there, consider the numbers associated with the two points as the x and y coordinates in the square.{{< /spoiler >}}
```

The problem is that there’s some ambiguity when it comes to the edges of this square. Remember, the endpoints $0$ and $1$ on the interval really correspond to the same point of the loop, so they need to be glued together if we are to faithfully map back to the loop.

So all the points on the left edge of the square really represent the same pair of points on the loop as the corresponding coordinates on the right edge of the square. If this square is going to represent pairs on the loop in a unique way, we’ll need to glue this left edge to the right edge.

Likewise, the bottom edge needs to be glued to the top edge, since the y-coordinates of $0$ and $1$ each really represent the same second point in a given pair of points on the loop.

Let’s mark each edge with some arrows to remember how the edges should be lined up:

{{< figure image="square-alignment-arrows.png" >}}
Matching points along opposite edges of the square both correspond to the same pair of points on the loop. To represent this faithfully, we need to glue together the edges of the square as indicated by the arrows.
{{< /figure >}}

> If you bend this square to perform the gluing, first rolling it into a cylinder to glue the left and right edges, then gluing the ends of the cylinder, which represent the top and bottom edges, we get a torus (better known as the surface of a doughnut).

Reviewer: ```suggestion
If you bend this square to perform the gluing, first rolling it into a cylinder to glue the left and right edges, then gluing the ends of the cylinder, which represent the top and bottom edges, we get a torus, the surface of a doughnut.
```

{{< figure image="roll-into-torus.png" video="roll-into-torus.mp4" >}}
When the square is wrapped up and its edges glued together following the arrows, it becomes a torus.
{{< /figure >}}

By gluing together the corresponding edges, we’ve solved the problem of duplicate mappings. Every individual point on this torus corresponds to a *unique* pair of points on the closed loop, and likewise every pair of points on the loop corresponds to a unique point on the torus.

{{< figure image="torus-and-loop-correspondence.png" >}}

> There is a unique correspondence between pairs of points on the loop and individual points on the torus.

Reviewer: Combine this image with the one below it, making this image the top of the next one.

If the torus is the natural shape for *ordered* pairs of points on the loop, what’s the natural shape for *unordered* pairs?

Remember, the whole reason we’re doing this is to show that two distinct pairs of points on the loop share a midpoint and are the same distance apart.  But if we consider the pair $(a, b)$ to be distinct from $(b, a)$, that would give us two separate pairs which trivially have the same midpoint and distance apart! That’s like saying you can always find a rectangle as long as you consider any pair of points to be a rectangle. Not helpful!

To find a surface representing *unordered* pairs, let’s look back at our unit square. We want the coordinates $(0.2, 0.3)$, say, to represent the same pair as $(0.3, 0.2)$. In general the coordinates $(x, y)$ should represent the same thing as $(y, x)$.

{{< figure image="matching-points.png" >}}
Since we’re interested in unordered pairs, the points $(x, y)$ and $(y, x)$ on the square both represent the same pair of points on the loop.
{{< /figure >}}

Once again, we capture this idea by gluing points together when they’re supposed to represent the same pair, which in this case requires folding the square diagonally.

> {{< figure image="fold-square.png" video="fold-square.mov" >}}

Reviewer: Again, for some reason, this clip does not show for me.  I wonder if it has anything to do with using .mov instead of .mp4?  It shouldn't...

{{< figure image="mobius-maps-to-surface.png" >}}
Points on the Möbius strip correspond to pairs of points on the loop which correspond to points on the 3D surface.
{{< /figure >}}

These two mappings combined give us a way to map points on the Möbius strip directly to points on the 3D surface. But what does this mapping look like, visually? Which points on the Möbius strip map to which points on the 3D surface?

To get a hint, consider the pairs of points $(X, X)$. Where do these points fall on the Möbius strip, and where do they fall on the 3D surface?

**Answer:** {{< spoiler >}}Pairs like $(X, X)$ correspond to points on the red edge of the Möbius strip. And for the 3D surface, the corresponding points are exactly on the loop. So when the Möbius strip is mapped onto this surface, it must be done in such a way that the edge of the strip maps onto the loop in the xy-plane.{{< /spoiler >}}

> {{< figure video="map-strip-to-surface.mov" >}}

Reviewer: Again, this doesn't play for me.

And that’s the proof!

{{< section >}}

# So Who Cares About Topology?
But of course, we haven’t *really* completed the proof. Our intuitive sense that the edge of the Möbius strip can’t be glued down to a 2D plane isn’t rigorous; it’s just something that *seems* true.

Perhaps you make a Möbius strip out of paper. No matter how you flex and bend the Möbius strip, placing the entire edge onto the surface of your table just isn’t going to work. But how do we prove that fact mathematically? How do we translate the intuitively, physically obvious into a mathematical model we can reason about formally?

If you try to prove that it’s impossible to flatten the edge of a Möbius strip into a 2D plane, you may stumble across the Sudanese Möbius band, which *does* in fact do just that! It’s a Möbius strip where the edge forms a perfect 2-dimensional circle.

> {{< figure image="sudanese-mobius-band.png" >}}

Reviewer: We may also want to link to this, which makes the whole thing a lot clearer, to me at least.  (Still confusing, but not _as_ confusing).

https://vimeo.com/286360639

If you understand the desire to turn your spatial intuitions into a rigorous air-tight line of reasoning, you understand the spirit of true topology. There is a popular conception of topology as a kind of rubber-sheet-geometry, where mathematicians dream up strange surfaces and spaces which they consider the same when you can stretch or morph one into the other without tearing it or gluing.

This is... well... kind of true, but it says nothing about why we’d care. The torus and Möbius strip arose for us as a way to represent and rephrase a completely separate non-topological problem. The mental picture of morphing without tearing or gluing is a way to conceptualize the idea of continuous one-to-one associations.

If you’re curious about which ideas from topology are useful to this problem, there are numerous theorems about which surfaces can and cannot be embedded into 3D space. While it’s possible to represent the Möbius strip and the torus in three-dimensions, there are other surfaces, like the Klein bottle[^3], which can only live in four or more dimensions, unless you allow for self-intersection.

> It turns out that gluing the edge of a Möbius strip to the edge of a disk gives us a surface which is topologically equivalent to something called the "real projective plane". So the surface we defined above for the inscribed rectangle problem, combined with the interior of the closed loop on the plane which it hugs, might be described by a mathematician as an embedding of the real projective plane into 3D space. And as you may have guessed, those mathematicians are a clever bunch, and they proved that you can’t have such an embedding. That trying to jam this surface into 3D space forces it to intersect itself.

Reviewer: ```suggestion
It turns out that gluing the edge of a Möbius strip to the edge of a disk gives us a surface which is topologically equivalent to something called the "real projective plane". So the surface we defined above for the inscribed rectangle problem, combined with the interior of the closed loop on the plane which it hugs, might be described by a mathematician as an embedding of the real projective plane into 3D space. And as you may have guessed, those mathematicians are a clever bunch, and they proved that you can’t have such an embedding. That is to say, trying to jam this surface into 3D space forces it to intersect itself.
```

> The proof is, as the lazy author loves to write, beyond the scope of this article. But perhaps by now you have an appreciation for why we’d ever care.

Reviewer: ```suggestion
If you're itching to understand _why_ this is true, wonderful! Consider this an excuse to start digging more into topology. Facts about which surfaces are topologically distinct from others, and how many dimensions they can live in, can be surprisingly hard to formalize and prove. But if you understand this proof sketch to the inscribed rectangle problem, you'll understand how these kinds of facts are more than mere idle ponderings.
```
